lib/resque/failure/redis.rb

```diff
@@ -48,14 +48,18 @@ def self.all(offset = 0, limit = 1, queue = nil)
       end
 
       def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil, order = 'desc')
+        if class_name
+          original_limit = limit
+          limit = count
```

Comment:
legit concern, but I'm just cherry-picking this (it has already been merged). I think if we have concerns about this code, it should be addressed outside of this PR (I can do that if you like, but not here I think).